### PR TITLE
First tag, untag THEN reload the image

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/image/image.go [hunk 1]
```diff
@@ -383,8 +386,11 @@ func (i *Image) UntagImage(tag string) error {
 			newTags = append(newTags, t)
 		}
 	}
+	if err := i.imageruntime.store.SetNames(i.ID(), newTags); err != nil {
+		return err
+	}
 	i.reloadImage()
```
Comment (mheon) [inline libpod/image/image.go]: To be safe, we may want to have one of these at the beginning of the function too, to grab an accurate list of names to set

Comment (baude) [reply libpod/image/image.go]: i thought of that too but you that reread has cost (locks, io) and therefore my feeling was if something modifies the image, it should reload it afterwards.

Comment (mheon) [reply libpod/image/image.go]: My concern is concurrency - two 'podman tag' launched at basically the same time in different terminals.

Comment (baude) [reply libpod/image/image.go]: fixed

**Comment by mheon** [comment]
LGTM
@rh-atomic-bot r+

**Pull request closed without merging**